libs/constrained-generators/src/Constrained/GenT.hs

```diff
@@ -339,23 +399,14 @@ frequency2 (n, g1) (m, g2)
 
 -- ======================================
 
--- | Temporarily extend the stack while executing 'm', and revert to the old stack if successful
-push :: forall m a. MonadGenError m => [String] -> GenT GE a -> GenT m a
-push [] m = dropGen m
-push (x : xs) m =
-  case explain (x NE.:| xs) m of
-    GenT f -> (GenT g)
-      where
-        g :: GenMode -> Gen (m a)
-        g mode = do
-          result <- f mode
-          case result of
-            Result (_ : ys) a -> pure $ runGE (Result ys a)
-            other -> pure $ runGE other
+-- | like explain for GenT, but uses [String] rather than (NonEmpty String)
+--   if the list is null, it becomes the identity
+push :: forall m a. MonadGenError m => [String] -> GenT m a -> GenT m a
```

Comment:
Why not get rid of `pushGE` and use one function for both use cases?
```suggestion
push :: forall m a. MonadGenError m => [String] -> m a -> m a
```

Reply:
What a good idea.